core/controllers/acl_decorators_test.py

```diff
@@ -6706,16 +6710,10 @@ def extract_request_message_vm_id_and_signature(
                 OppiaMLAuthInfo. Message at index 0, vm_id at index 1 and
                 signature at index 2.
             """
-            # Here we use cast because we are narrowing down the type of
-            # 'normalized_payload' from Dict[str, Any] to a particular
-            # TypedDict that was defined according to the schemas. So that
-            # the type of fetched values is not considered as Any type.
-            payload = cast(
-                MockHandlerNormalizedPayloadDict, self.normalized_payload
-            )
-            signature = payload['signature']
-            vm_id = payload['vm_id']
-            message = payload['message']
+            assert self.normalized_payload is not None
```

Comment:
Same question here as in the other PR (about why we need to do this); ditto elsewhere.

Reply:
I think we should keep the conversation in only one place https://github.com/oppia/oppia/pull/16461#discussion_r1011443932 and mention the conclusion of that conversation here.